src/profile-logic/tracks.js

```diff
@@ -932,9 +1029,36 @@ export function computeDefaultVisibleThreads(
     return new Set(profile.meta.initialVisibleThreads);
   }
 
+  const allTrackThreads = new Set();
+  for (const globalTrack of tracksWithOrder.globalTracks) {
+    switch (globalTrack.type) {
+      case 'process':
+        if (globalTrack.mainThreadIndex !== null) {
+          allTrackThreads.add(globalTrack.mainThreadIndex);
+        }
+        break;
+      default:
+        break;
+    }
+  }
+
+  // $FlowExpectError Flow doesn't know about Array.prototype.flat.
+  const localTracks = Array.from(
+    tracksWithOrder.localTracksByPid.values()
+  ).flat();
+  for (const localTrack of localTracks) {
```

Comment:
It could be more performant to nest 2 for loops instead of looping over all local tracks several times, what do you think? I'm thinking about these profiles where we have so many tracks.